File: code/java/src/main/java/mediator/Tester.java
package mediator;

class Tester extends Colleague {
    public Tester(Mediator mediator) {
        super(mediator);
    }

    @Override
    public void send(String message) {
        System.out.println("Tester sending message: " + message);
        mediator.request(message, this);
    }

    @Override
    public void receive(String message) {
        System.out.println("Tester received task: " + message);
    }
}

File: code/typescript/mediator/Tester.ts
import { Colleague } from "./Colleague";
import { Mediator } from "./Mediator";

export class Tester extends Colleague {
    constructor(mediator: Mediator) {
        super(mediator);
    }

    send(message: string): void {
        console.log(`Tester sending message: ${message}`);
        this.mediator.request(message, this);
    }

    receive(message: string): void {
        console.log(`Tester received task: ${message}`);
    }
}
